src/archivematicaCommon/lib/archivematicaFunctions.py

```diff
@@ -536,6 +536,17 @@ def relative_path_to_aip_mets_file(uuid, current_path):
     return mets_path
 
 
+def filter_replicated_aips(aips):
+    """Filter out replicated AIPs.
+
+    :param aips: List of package info for AIPs returned by Storage
+    Service.
+
+    :returns: Filtered AIP list.
+    """
+    return [aip for aip in aips if aip["replicated_package"] is None]
```

Comment:
This might be a useful helper in `AMClient` where we already have a helper for compressed packages as an example. I am not convinced it sits particularly well here as its own helper in Archivematica given it is just a single list comprehension. 

As this is dealing with storage service functions, a nice approach might be to modify the storage service helper. 

* `storageService.get_file_info(package_type="AIP")`

If you were interacting with an API, you might see a filter. Something like:

* `storageService.get_file_info(package_type="AIP", filter_replicas=True)`
* `storageService.get_file_info(package_type="AIP", filter=REPLICAS)` (where replicas is an `enum` like value.

Reply:
It dawns on me too, is there a way to only filter once? What information are we getting from the `storage_service.get_file_info()`? 

Is it possible to:

1. Retrieve all AIPs and AICs.
2. Filter replicated packages.
3. Direct the workflow in these functions by testing here whether something is an AIC or an AIP?

Reply:
Hi @ross-spencer - thanks, this is good insight. I've taken the spirit of that you've said here and tried to address it in commit 7cca2c9624865044c95e160104a729fe01777361, which makes a single call to the storage service for packages, does all of the filtering in one shot (and moves that helper to `storageService.py`, where I think it's a better fit), and manages to remove a fair bit of duplicative code in the process. Let me know what you think.